app/controllers/invites_controller.rb

```diff
@@ -22,8 +22,17 @@ def show
     invite = Invite.find_by(invite_key: params[:id])
     if invite.present? && invite.redeemable?
       if current_user
-        added_to_group = false
+        invited_user = InvitedUser.find_or_initialize_by(user: current_user, invite: invite)
+        if invited_user.new_record?
+          invited_user.save!
+          Invite.increment_counter(:redemption_count, invite.id)
+          invite.invited_by.notifications.create!(
+            notification_type: Notification.types[:invitee_accepted],
+            data: { display_username: current_user.username }.to_json
+          )
```

Comment:
I'm thinking it might be safer to wrap these 3 DB operations into a transaction so that we don't end up in a weird state with the counter and notifications.